rust/lance-encoding/src/encodings/physical/rle.rs

```diff
@@ -0,0 +1,775 @@
+            }
+
+            let values_buffer = LanceBuffer::Owned(values_bytes);
+            let lengths_buffer = LanceBuffer::Owned(lengths_bytes);
+
+            let log_num_values = if is_last_chunk {
+                0
+            } else {
+                assert!(
+                    values_processed.is_power_of_two(),
+                    "Non-last chunk must have power-of-2 values"
+                );
+                values_processed.ilog2() as u8
+            };
+
+            let chunk = MiniBlockChunk {
+                buffer_sizes: vec![(num_runs * bytes_per_value) as u16, num_runs as u16],
+                log_num_values,
+            };
+
+            all_buffers.push(values_buffer);
+            all_buffers.push(lengths_buffer);
+            chunks.push(chunk);
+
+            offset += values_processed;
+            values_remaining -= values_processed;
+        }
+
+        Ok((all_buffers, chunks))
+    }
+
+    /// Encodes a chunk of data using RLE compression with dynamic boundary detection.
+    ///
+    /// This function processes values sequentially, detecting runs (sequences of identical values)
+    /// and encoding them as (value, length) pairs. It dynamically determines whether this chunk
+    /// should be the last chunk based on how many values were processed.
+    ///
+    /// # Key Features:
+    /// - Tracks byte usage to ensure we don't exceed MAX_MINIBLOCK_BYTES
+    /// - Maintains power-of-2 checkpoints for non-last chunks
+    /// - Splits long runs (>255) into multiple entries
+    /// - Dynamically determines if this is the last chunk
+    ///
+    /// # Returns:
+    /// - values_bytes: Buffer containing unique run values
+    /// - lengths_bytes: Buffer containing run lengths (u8)
+    /// - num_runs: Number of runs encoded
+    /// - values_processed: Number of input values processed
+    /// - is_last_chunk: Whether this chunk processed all remaining values
+    fn encode_chunk_rolling<T>(
+        &self,
+        data: &LanceBuffer,
+        offset: usize,
+        values_remaining: usize,
+    ) -> (Vec<u8>, Vec<u8>, usize, usize, bool)
+    where
+        T: bytemuck::Pod + PartialEq + Copy + std::fmt::Debug,
+    {
+        let type_size = std::mem::size_of::<T>();
+        let data_slice = data.as_ref();
+
+        let chunk_start = offset * type_size;
+        let max_by_count = MAX_MINIBLOCK_VALUES as usize;
+        let max_values = values_remaining.min(max_by_count);
+        let chunk_end = chunk_start + max_values * type_size;
+
+        if chunk_start >= data_slice.len() {
+            return (Vec::new(), Vec::new(), 0, 0, false);
+        }
+
+        let chunk_data = &data_slice[chunk_start..chunk_end.min(data_slice.len())];
+        let typed_data: &[T] = bytemuck::cast_slice(chunk_data);
+
+        if typed_data.is_empty() {
+            return (Vec::new(), Vec::new(), 0, 0, false);
+        }
+
+        let estimated_runs = (max_values / 10).max(1).min(max_values);
```

Comment:
Why `.min(max_values)`?  It doesn't seem like `(max_values / 10).max(1)` could be larger than `max_values`?

Reply:
Fixed.